keyboards/dz60/dz60.h

```diff
@@ -11,13 +11,13 @@
 	K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, K014, \
 	K100,       K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, \
 	K200,       K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K213,       \
-	K300, K301, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311,       K313, K314, \
+	K300, 		K302, K303, K304, K305, K306, K307, K308, K309, K310, K311,	K312,  K313, K314, \
```

Comment:
These changes in dz60.h might actually break other people's keymaps. Notably
Ansi_plus_fn_arrows, LEdiodes, atlacat, default, and a few others. By removing K301 but adding K312 you still ensure that the keymaps they chose compile with the LAYOUT, but it might not work as they want on their board now.

I suggest you keep the original LAYOUT, and make your own LAYOUT macro. Or look at the other LAYOUT macros and see which one best suites your needs.

Reply:
I don't really understand what do you mean by "LAYOUT macro" and how to do that. Obviously who hasn't 2u left shift it won't work how do they want. So the question is do i need to make the dz60.h to work for everyone even if they use 2.25 left shift ? Or do i need to make another dz60.h especially for my layout ? Either way, i don't know how to solve this.

Reply:
This is a LAYOUT macro:

```c
#define LAYOUT( \
	K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, K014, \
	K100,       K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, K114, \
	K200,       K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, K212, K213,       \
	K300,       K302, K303, K304, K305, K306, K307, K308, K309, K310, K311,	K312, K313, K314, \
	K400, K401,       K403, K404,       K406,       K408,       K410, K411, K412, K413, K414  \
) { \
	{ K000,  K001,  K002,  K003,  K004,  K005,  K006,  K007,  K008,  K009,  K010,  K011,  K012,  K013,  K014 }, \
	{ K100,  KC_NO, K102,  K103,  K104,  K105,  K106,  K107,  K108,  K109,  K110,  K111,  K112,  K113,  K114 }, \
	{ K200,  KC_NO, K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  K211,  K212,  K213,  KC_NO }, \
	{ K300,  KC_NO,  K302,  K303,  K304,  K305,  K306,  K307,  K308,  K309,  K310,  K311, K312,	 K313,  K314 }, \
	{ K400,  K401,  KC_NO, K403,  K404,  KC_NO, K406,  KC_NO, K408,  KC_NO, K410,  K411,  K412,  K413,  K414 }  \
}
```

The top half represents the physical keys on the keyboard, and the bottom half designates where each key is in the switch matrix (rows and columns).

The change you've made to `dz60.h` won't just affect your keymap, but every keymap that uses that layout macro. What this means is that your keymap will work fine, but the keymaps of every other user whose keymap uses that macro will break. Specifically, it moves the keys around for other users, such that a user who updates their keymap, and whose keymap uses this macro may end up with misplaced keys, and may have to do something like use the X key to type a Z or something.

Reply:
@noroadsleft i understand perfectly what you say and the problem. I actually after reading your answer, got back to @mechmerlin message and read it more a few times and i think i now what he was referring to now. I have to see if there (in dz60.h file) is existing LAYOUT that suits my actual layout and if there is not to edit it and add one that will be compatible with my layout. I will try to fix it. Thank you

Reply:
Adding on to this to reference our discussion on discord.

It seems that @rundeploy's layout most closely matches LAYOUT_directional, however LAYOUT_directional omits K311. So, the LAYOUT_directional layout might have actually been wrong all this time. 

Conclusion was to try out LAYOUT_all and then explore LAYOUT_directional.